bindings.c

```diff
@@ -4212,6 +4220,212 @@ static int proc_swaps_read(char *buf, size_t size, off_t offset,
 	return rv;
 }
 
+/*
+ * Find the process pid from cgroup path.
+ * eg:from /sys/fs/cgroup/cpu/docker/containerid/cgroup.procs to find the process pid.
+ * pid_buf : put pid to pid_buf.
+ * dpath : the path of cgroup. eg: /docker/containerid or /docker/containerid/child-cgroup ...
+ * depth : the depth of cgroup in container.
+ * sum : return the number of pid .
+ * cfd : the file descriptor of the mounted cgroup. eg: /sys/fs/cgroup/cpu
+ */
+static int calc_pid(char ***pid_buf, char *dpath, int depth, int sum, int cfd)
+{
+	DIR *dir;
+	int fd;
+	struct dirent *file;
+	/* path = dpath + "/cgroup.procs" + /0 */
+	char *path = (char *)malloc(strlen(dpath) + 20);
+
+	strcpy(path, dpath);
+	fd = openat(cfd, path, O_RDONLY);
+	if (fd < 0)
+		return sum;
+
+	if ((dir = fdopendir(fd)) == NULL) {
+		return sum;
+	} else {
```

Comment:
Since you returned on failure, no point having an } else { and
subsequent block indented.